src/endpoint/s3/ops/s3_get_object_attributes.js

```diff
@@ -0,0 +1,110 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../../util/debug_module')(__filename);
+const s3_utils = require('../s3_utils');
+const http_utils = require('../../../util/http_utils');
+const S3Error = require('../s3_errors').S3Error;
+
+/**
+ * http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGETacl.html
+ */
+async function get_object_attributes(req, res) {
+    const version_id = s3_utils.parse_version_id(req.query.versionId);
+    const encryption = _parse_encryption_headers(req);
+    const attributes = _parse_attributes(req);
+
+    const params = {
+        bucket: req.params.bucket,
+        key: req.params.key,
+        version_id: version_id,
+        encryption: encryption, // GAP - we don't use it currently
+        md_conditions: http_utils.get_md_conditions(req), // GAP - we don't use it currently in all namespaces (for example - not in NSFS)
+        attributes: attributes,
+    };
+    dbg.log2('params after parsing', params);
+    const reply = await req.object_sdk.get_object_attributes(params);
```

Comment:
I dont understand why we need a new function in the namespaces and not just call read_object_md.

Reply:
I imitate the behavior of `get_object_acl` which is implemented in every namespace and calls `this.read_object_md` from inside. I thought that since the suggested implementation is partial, and we might want to expand it in the future, this would give us more flexibility. The list of params is not the same between them and inside the function in each namespace, I delete what is not relevant for read_object_md (doing it in the main file `s3_get_object_attributes.js` might be confusing, and encapsulating it in the inside implementation of every namespace seems like more appropriate).

Reply:
doesn't seem needed to me. just call read_object_md and return whatever is needed from the op. deleting params.attributes is not needed if you dont pass it in the first place.

Reply:
wait I forgot - you did that for the list parts option right? in that case I would have object_sdk handle it and check if the namespace has its own implementation of get_object_attributes function and call it if it does (for example for ns-s3) but will fallback to calling get_object_md and return just atteibutes that are in md for any other namespace

Reply:
We added the fallback and only implemented namespace s3.